plugins/RnD/ColorMatching.py

```diff
@@ -0,0 +1,1167 @@
+0.9357202016619977,
+0.9326190350739556,
+0.9300196714424732,
+0.927444559208246,
+0.9252602575622887,
+0.9229621052048547,
+0.9171357616783294,
+0.9120024753827068,
+]
+
+COLOR_MODEL_ALIASES = {
+                        'CIE 1931 2 deg' : 'CIE 1931 2 Degree Standard Observer',
+                        'CIE 1931 10 deg' : 'CIE 1964 10 Degree Standard Observer',
+                        'CIE 2015 2 deg' : 'CIE 2015 2 Degree Standard Observer',
+                        'CIE 2015 10 deg' : 'CIE 2015 10 Degree Standard Observer'
+}
+
+class ColorMatching(EnlightenPluginBase):
+
+        def disconnect(self):
+            super().disconnect()
+
+        def get_configuration(self):
+
+            self.name = f"ColorMatching {COLOR_PLUGIN_VERSION}"
+            self.has_other_graph = False
+            self.auto_enable = True
+            self.is_blocking = True
+            self.block_enlighten = False
+
+            self.field(
+                name="Color Model",
+                datatype="combobox",
+                direction="input",
+                choices=list(COLOR_MODEL_ALIASES.keys()),
+                tooltip="Color model to use",
+            )
+
+            self.field(
+                name="XYZ",
+                datatype=str,
+                direction="output",
+                tooltip="XYZ",
+            )
+
+            self.field(
+                name="RGB",
+                datatype=str,
+                direction="output",
+                tooltip="RGB",
+            )
+
+            self.field(
+                name="sRGB",
+                datatype=str,
+                direction="output",
+                tooltip="sRGB",
+            )
+
+            self.field(
+                name="Color Temp",
+                datatype=str,
+                direction="output",
+                tooltip="Color temperature",
+            )
+
+            self.field(name="Apply Calibration",
+                       direction="input",
+                       datatype=bool,
+                       initial=True,
+                       expert=False,
+                       tooltip="Apply hard coded (relative) spectral intensity calibration")
+
+            log.debug(f"{self.name} started")
+
+        def process_request(self, request):
+            """ Process new spectrum received from ENLIGHTEN
+
+                Spectra are divided by a hard-coded calibration factor (if enabled) to give a flat spectral response,
+                interpolated to 1 nm wavelength spacing (necessary for the color-science package), then converted
+                into CIE XYZ tristimulus values.
+
+                CIE XYZ is the necessary basis for conversion to other colorspaces.
+
+                The function demonstrates conversion to sRGB, RGB and color temperature.
+
+            """
+            wavelength = np.array(self.settings.wavelengths, dtype=np.float64)
+            spectrum = np.array(request.processed_reading.processed, dtype=np.float64)
```

Comment:
Consider these for better handling of ProcessedReadings that have already been cropped for horizontal ROI, and/or interpolated by ENLIGHTEN:
```
wavelength = np.array(request.processed_reading.get_wavelengths(), dtype=np.float64)
spectrum   = np.array(request.processed_reading.get_processed(), dtype=np.float64)
```

Note that this would require the calibration array to be similarly altered, as it's in pixel space.

For now, I'd probably just verify that len(spectrum) == len(calibration) before performing the division.

Reply:
Good call.  Updated to interpolate the calibration factor to the input wavelength range.  Therefore, cropped or otherwise altered spectra will be correctly handled.